src/test/java/com/squareup/javapoet/MethodSpecTest.java

```diff
@@ -0,0 +1,34 @@
+package com.squareup.javapoet;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public final class MethodSpecTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  @Test public void nullAnnotationsAddition() {
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage("method annotations must not be null.");
+    MethodSpec.methodBuilder("doSomething").addAnnotations(null);
+  }
+
+  @Test public void nullTypeVariablesAddition() {
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage("method type variables must not be null.");
+    MethodSpec.methodBuilder("doSomething").addTypeVariables(null);
+  }
+
+  @Test public void nullParametersAddition() {
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage("parameters must not be null.");
+    MethodSpec.methodBuilder("doSomething").addParameters(null);
+  }
+
+  @Test public void nullExceptionsAddition() {
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage("exceptions must not be null.");
+    MethodSpec.methodBuilder("doSomething").addExceptions(null);
+  }
+}
```

Comment:
Not sure why checkstyle isn't yelling at you here. I thought we had it configured to enforce trailing newlines.